src/averageAges.js

```diff
@@ -15,11 +15,23 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let average = 0;
+
+  let menList = people.filter(person => person.sex === 'm');
+
+  if (arguments.length === 2) {
+    menList = people.filter(person => (
+      person.sex === 'm' && century === Math.ceil(person.died / 100)
+    ));
+  }
+
+  const yearsOfLife = menList.map(person => person.died - person.born);
+
+  const sumOfYears = yearsOfLife.reduce((acc, number) => acc + number, 0);
```

Comment:
You can combine these two methods into one reduce.
```suggestion
  const sumOfYears = menList.reduce((sumAges, person) => {
  const personAges = person.died - person.born;
  
  return sumAges + personAges;
  }, 0);
```
And I see that you use the same methods in another function, as an idea, you can move this logic into a separate function and then just call this function here with the passed array.